datahub-web-react/src/app/entity/shared/EntityDropdown/CreateGlossaryEntityModal.tsx

```diff
@@ -82,12 +101,21 @@ function CreateGlossaryEntityModal(props: Props) {
                     refetch();
                     if (isInGlossaryContext) {
                         // either refresh this current glossary node or the root nodes or root terms
-                        const nodeToUpdate = entityData?.urn || getGlossaryRootToUpdate(entityType);
+                        const nodeToUpdate = props.clone
+                            ? getGlossaryRootToUpdate(entityType)
+                            : entityData?.urn || getGlossaryRootToUpdate(entityType);
```

Comment:
so i think you may have unconvered an existing bug here that can be fixed and work for both the clone case as well as all regular cases. instead of always updating the root when cloning, I think this line can be:
`const nodeToUpdate = selectedParentUrn || getGlossaryRootToUpdate(entityType);` because it's the parent that needs updating in the sidebar if there is a parent, otherwise we update the root